services/outputhost/ackmanager.go

```diff
@@ -326,17 +331,17 @@ func (ackMgr *ackManager) acknowledgeMessage(ackID AckID, seqNum ackIndex, addre
 		} else {
 			if ackMgr.cgCache.cachedCGDesc.GetOwnerEmail() == SmartRetryDisableString {
 				ackMgr.logger.WithFields(bark.Fields{
-					`Address`:     address,
-					`addr`:        msg.addr,
-					common.TagSeq: seqNum,
-					`isNack`:      isNack,
+					`Address`: address,
+					`addr`:    msg.addr,
+					`index`:   index,
+					`isNack`:  isNack,
 				}).Info(`msg ack`)
 			}
 			if !isNack {
 				msg.acked = true // This is the only place that this field of msg is changed. It was initially set under a write lock elsewhere, hence we can have a read lock
 				// update the last acked sequence, if this is the most recent ack
-				if ackMgr.lastAckedSeq < common.SequenceNumber(seqNum) {
-					ackMgr.lastAckedSeq = common.SequenceNumber(seqNum)
+				if ackMgr.lastAckedSeq < msg.seqnum {
+					ackMgr.lastAckedSeq = msg.seqnum
```

Comment:
@kirg was there a bug on the left? `seqNum` is supposed to be `ackIndex`. It shouldn't be cast into `common.SequenceNumber`.